db.go

```diff
@@ -677,7 +692,7 @@ func (a *dbAppender) appenderAt(t int64) (*metaAppender, error) {
 	}
 
 	var hb headBlock
-	for _, h := range a.db.appendable() {
+	for _, h := range a.db.appendableHeads() {
```

Comment:
This section of `appenderAt` will only be executed the first time we have to figure out the appropriate head block for timestamp `t`. In the general case, we return within the for loop.